### PR TITLE
Add check for bad pool size when adding pools with calicoctl

### LINKED ISSUE #505: pools with ranges don't work well with IPAM
`calicoctl pool range add ...` creates a bunch of small pools that add up to the range identified.  This is a problem if Calico IPAM is used, since we use a fixed-size block, so it will ignore any pools smaller than the block size (64-addresses in latest version).

In the worst case, if no pools are the block size or larger, we will completely fail to allocate addresses.

c.f. error logs in Kubernetes: http://pastebin.com/PGMe67Ws

### PR DESCRIPTION
**Pull request by TrimBiggs**
Fixes #505

### CODE REVIEW

File: calicoctl/calico_ctl/pool.py [hunk 1]
```diff
@@ -151,8 +162,18 @@ def ip_pool_range_add(start_ip, end_ip, version, ipip, masquerade):
             sys.exit(1)
 
     cidrs = netaddr.iprange_to_cidrs(start_ip, end_ip)
+    new_pools = []
     for ip_net in cidrs:
-        new_pool = IPPool(ip_net.cidr, ipip=ipip, masquerade=masquerade)
+        try:
+            new_pools.append(IPPool(ip_net.cidr, ipip=ipip, masquerade=masquerade))
+        except CidrTooSmallError:
+            print "IPv%s range must start and end on a %s prefix network." \
```
Comment (robbrockbank) [inline calicoctl/calico_ctl/pool.py]: I think this error message needs a little work.  I'm wondering if we should make it more transparent what this command does - i.e. inform the user that we are breaking the range into CIDRs and storing as CIDRs.  If we don't then the comment about the offending pool would seem a little odd.

Not sure the commend about starting and ending on a /26 prefix network is true.  I think it is more that the start and end ip address must have at least the lower n bits unset.  I've no idea what the terminology would be for that - although you could talk about an individual pool being too small.

Comment (TrimBiggs) [inline calicoctl/calico_ctl/pool.py]: How about:
```
"The IP range is split into multiple CIDRs using prefix length of '/26' or lower.  When formed into CIDRs, the range specified includes %s with prefix length %s, which is too large.  No pools were added."
```

Comment (TrimBiggs) [inline calicoctl/calico_ctl/pool.py]: Or possibly displaying all pools that would be added.

Comment (robbrockbank) [inline calicoctl/calico_ctl/pool.py]: I'm happy with the pools being displayed

**Comment by robbrockbank** [comment]
lgtm

**Pull request merged**